Documentation/hacking.md

```diff
@@ -10,24 +10,46 @@ For more information on the rkt internals, see the [`devel`](devel/) documentati
 rkt should be able to be built on any modern Linux system.
 For the most part the codebase is self-contained (e.g. all dependencies are vendored), but assembly of the stage1 requires some other tools to be installed on the system.
 
-### Build-time requirements
-
-* Linux 3.8+
-  * autoconf
-  * automake
-  * bc
-  * cpio
-  * gcc
-  * glibc development and static pieces (on Fedora/RHEL/Centos: glibc-devel and glibc-static packages, on Debian/Ubuntu libc6-dev package)
-  * gpg
-  * libacl development files (on Fedora/RHEL/Centos: libacl-devel, on Debian/Ubuntu libacl1-dev)
-  * make
-  * openssl
-  * patch
-  * realpath
-  * squashfs-tools
-  * wget
-* Go 1.4+
+### Requirements
+
+The following list contains both build-time and run-time requirements. Run-time requirement are mentioned explicitly.
+
+* GNU Make
+* Go 1.4+ (ideally 1.5.3 or later)
+* Linux 3.8+ (ideally 4.3+ at run-time to have overlay-on-overlay working)
+  * at run-time, following options are required or nice-to-have:
+    * CONFIG_CGROUPS
+    * CONFIG_NAMESPACES
+    * CONFIG_UTS_NS
+    * CONFIG_IPC_NS
+    * CONFIG_PID_NS
+    * CONFIG_NET_NS
+    * CONFIG_OVERLAY_FS (nice to have)
+* autoconf
+* aclocal (usually a part of automake)
+* bash
+  * used during build
+  * also a run-time requirement for the host stage1 image flavor
+* glibc
+  * a dynamic library for rkt
+  * development headers and a static library (when building any stage1 image flavor)
+* gofmt (usually distributed with Go)
+* govet (usually distributed with Go)
+* TrouSerS (only when TPM is enabled)
+* libdl (when building any stage1 image flavor)
```

Comment:
I was differentiating between a common situation (development headers required during build, a binary links to a library, like libdl) and something else like (development headers required during build, a binary does not link to a library, like libacl).